src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -368,65 +369,80 @@ where
 
         // Execute multiple statements in simple query. KISS later.
         for stmt in stmts {
-            let session = session.clone();
-            // execute query
-            let res = session.clone().run_one_query(stmt, Format::Text).await;
-            for notice in session.take_notices() {
-                self.stream
-                    .write_no_flush(&BeMessage::NoticeResponse(&notice))?;
-            }
-            let mut res = res.map_err(|err| PsqlError::QueryError(err))?;
+            let span = tracing::info_span!("run_stmt", %stmt);
 
-            for notice in res.notices() {
-                self.stream
-                    .write_no_flush(&BeMessage::NoticeResponse(notice))?;
-            }
+            self.inner_process_query_msg_one_stmt(stmt, session.clone())
+                .instrument(span)
+                .await?;
```

Comment:
The changes here are only to extract the codes in the `for` block and instrument it with a span.